tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Configuration/GlobalConfigurationProvider.cs

```diff
@@ -59,5 +59,50 @@ public string GetApplicationName()
 
             return applicationName;
         }
+
+        private object maxRequestsPerPeriodLock = new object();
+        private int maxRequestsPerPeriod = -1;
+
+        public int GetMaxRequestsPerPeriod()
+        {
+            if (maxRequestsPerPeriod == -1)
+            {
+                lock (maxRequestsPerPeriodLock)
+                {
+                    if (maxRequestsPerPeriod == -1)
+                    {
+                        ConfigurationSetting applicationNameSetting = configurationClient.GetConfigurationSetting(
```

Comment:
This probably still scales fine for now, but at some point we'll likely want to switch to getting multiple settings from the config store at once (assuming this is making a network call on each request). 
Perhaps in the future could switch to using [configuration providers](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1)? [Azure App Configuration](https://docs.microsoft.com/en-us/azure/azure-app-configuration/quickstart-aspnet-core-app?tabs=core3x#connect-to-an-app-configuration-store) appears to have built in support.

Reply:
Yeah ... each time I look at the configuration provider API I feel like I'm going down a rabbit hole ;)